toolkit/tools/imagecustomizer/docs/configuration.md

```diff
@@ -0,0 +1,539 @@
+# Mariner Image Customizer configuration
+
+Top-level type: [Config](#config)
+
+Operation ordering:
+
+1. Override the `/etc/resolv.conf` file with the version from the host OS.
+
+2. Update packages:
+
+   1. Remove packages ([PackageListsRemove](#packagelistsremove-string),
+   [PackagesRemove](#packagesremove-string))
+
+   2. Update base image packages ([UpdateBaseImagePackages](#updatebaseimagepackages-bool)).
+
+   3. Install packages ([PackageListsInstall](#packagelistsinstall-string),
+   [PackagesInstall](#packagesinstall-string))
+
+   4. Update packages ([PackageListsUpdate](#packagelistsupdate-string),
+   [PackagesUpdate](#packagesupdate-string))
+
+3. Update hostname. ([Hostname](#hostname-string))
+
+4. Copy additional files. ([AdditionalFiles](#additionalfiles-mapstring-fileconfig))
+
+5. Add/update users. ([Users](#users-user))
+
+6. Enable/disable services. ([Services](#services))
+
+7. Run post-install scripts. ([PostInstallScripts](#postinstallscripts-script))
+
+8. Run post-finalize scripts. ([FinalizeImageScripts](#finalizeimagescripts-script))
+
+9. Delete `/etc/resolv.conf` file.
+
+Note: The `/etc/resolv.conf` file is overridden so that the package installation and
+customization scripts can have access to the network.
+It is assumed there is a process that runs on boot that will set the `/etc/resolv.conf`.
+For example, `systemd-resolved`.
+Hence why the `/etc/resolv.conf` file is simply deleted at the end instead of being
+restored to its original contents.
+
+## Config
+
+The top-level type of the configuration.
+
+### SystemConfig [[SystemConfig](#systemconfig)]
+
+Contains the configuration options for the OS.
+
+```yaml
+SystemConfig:
+  Hostname: example-image
+```
+
+## FileConfig
+
+Specifies options for placing a file in the OS.
+
+Type is used by: [AdditionalFiles](#additionalfiles-mapstring-fileconfig)
+
+### Path [string]
+
+The absolute path of the destination file.
```

Comment:
maybe add a note that same file can have multiple destinations?

Reply:
That is a feature of the `AdditionalFiles` property, not the `FileConfig` object,